modules/afmongodb/afmongodb-legacy-uri.c

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright (c) 2010-2016 Balabit
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "afmongodb-legacy-uri.h"
+#include "afmongodb-legacy-private.h"
+
+#define DEFAULTHOST "127.0.0.1"
+#define SOCKET_TIMEOUT_FOR_MONGO_CONNECTION_IN_MILLISECS 60000
+
+static gboolean
+_parse_addr(const char *str, char **host, gint *port)
+{
+  if (!host || !port)
+    {
+      msg_debug("Host or port reference should not be NULL");
+      return FALSE;
+    }
+  char *proto_str = g_strdup_printf("mongodb://%s", str);
+  mongoc_uri_t *uri = mongoc_uri_new(proto_str);
+  if (!uri)
```

Comment:
I know that you prefer declaring variables inline, one feedback: long type names with qualifiers clutter up lines, making them more difficult to read.

I know why it is a good idea, local declarations and ask that stuff, but I really had to scroll sideways, and not once during this review.

If only we had type inference, we could use a short statement (like auto as in c++) but without that Iam not sure anymore that it is a good idea.

This case is not the worst offender, the append legacy host function is a lot worse.